### PR TITLE
Fix bug where a manually added node breaks the node list page

### LINKED ISSUE #9955: Cannot see Nodes in Explorer if using Dummy Node
Internal Reference: SURE-7126

**Adding a dummy node to a cluster breaks the node view in Cluster Explorer**

To Reproduce:

In the local cluster, import the following yaml:

```
apiVersion: v1
kind: Node
metadata:
  name: bigip1
annotations:
  flannel.alpha.coreos.com/public-ip: "10.10.168.2"
  flannel.alpha.coreos.com/backend-data: '{"VtepMAC":"00:0a:49:ce:c3:82"}'
  flannel.alpha.coreos.com/backend-type: "vxlan"
  flannel.alpha.coreos.com/kube-subnet-manager: "true"
spec:
  podCIDR: "10.42.255.0/24"
```

Go the nodes view - the page breaks with `Cannot read properties of undefined (reading 'cpu')`

### PR DESCRIPTION
**Pull request by nwmac**
Fixes #9955

Fixes bug where some manually added nodes can break the node list page.

Automated e2e test included

### CODE REVIEW

File: cypress/e2e/tests/pages/explorer/node-list.spec.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    cy.contains('.title > h1', 'Nodes').should('be.visible');
+
+    const nodeList = new BaseResourceList(cy.get('[data-testid="cluster-list-container"]'));
+
+    // Wait for loading indicator to go
+    cy.get('.data-loading').should('not.exist');
+
+    // Check table has 2 tows
+    nodeList.resourceTable().sortableTable().rowElements({ timeout: 2500 }).should((rows: any) => {
```
Comment (richard-cox) [inline cypress/e2e/tests/pages/explorer/node-list.spec.ts:37]: There's `checkRowCount` in cypress/e2e/po/components/sortable-table.po.ts. Was the issue with the longer timeout?

File: cypress/e2e/tests/pages/explorer/node-list.spec.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      expect(rows).not.to.equal(undefined);
+      expect(rows).to.have.length(2);
+    });
+
+    // Check the node names
+    nodeList.self().find('td.col-link-detail > span > a').then((links: any) => {
```
Comment (richard-cox) [inline cypress/e2e/tests/pages/explorer/node-list.spec.ts:43]: cypress/e2e/po/components/sortable-table.po.ts `rowNames`  should give you the same as this

**Review by richard-cox — CHANGES_REQUESTED** [review]
Functional side looks good, just some e2e things

File: cypress/e2e/tests/pages/explorer/node-list.spec.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import ClusterDashboardPagePo from '@/cypress/e2e/po/pages/explorer/cluster-dashboard.po';
+import HomePagePo from '@/cypress/e2e/po/pages/home.po';
+import ProductNavPo from '@/cypress/e2e/po/side-bars/product-side-nav.po';
+import BaseResourceList from '@/cypress/e2e/po/lists/base-resource-list.po';
+import { dummyNode } from '@/cypress/e2e/blueprints/explorer/nodes';
+
+describe('Nodes list', { tags: ['@adminUser'], testIsolation: 'off' }, () => {
```
[the remark this answers is not in the document]
Comment (nwmac) [reply cypress/e2e/tests/pages/explorer/node-list.spec.ts]: Done

File: cypress/e2e/tests/pages/explorer/node-list.spec.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  before(() => {
+    cy.login();
+    HomePagePo.goTo();
+
+    // Add dummy node that used ot cause a problem
+
+    // cy.deleteRancherResource('v1', 'nodes', 'bigip1');
```
[the remark this answers is not in the document]
Comment (nwmac) [reply cypress/e2e/tests/pages/explorer/node-list.spec.ts]: Removed

File: cypress/e2e/tests/pages/explorer/node-list.spec.ts (hunk 1 above)
Comment (nwmac) [reply cypress/e2e/tests/pages/explorer/node-list.spec.ts:36]: That didn't work for me - I think we have issues with that to do with the chaining and retries. I didn't want to change what we already had.

File: cypress/e2e/tests/pages/explorer/node-list.spec.ts (hunk 2 above)
Comment (nwmac) [reply cypress/e2e/tests/pages/explorer/node-list.spec.ts:42]: That didn't work for me - I think we have issues with that to do with the chaining and retries. I didn't want to change what we already had. There's something strange going on we could look at.

Comment (nwmac) [reply cypress/e2e/tests/pages/explorer/node-list.spec.ts]: Done - needed to use a string selector

File: cypress/e2e/tests/pages/explorer/node-list.spec.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    const nav = new ProductNavPo();
+
+    nav.navToSideMenuEntryByLabel('Nodes');
+
+    cy.contains('.title > h1', 'Nodes').should('be.visible');
+
+    const nodeList = new BaseResourceList(cy.get('[data-testid="cluster-list-container"]'));
+
+    // Wait for loading indicator to go
+    cy.get('.data-loading').should('not.exist');
```
[the remark this answers is not in the document]
Comment (nwmac) [reply cypress/e2e/tests/pages/explorer/node-list.spec.ts]: Done

**Review by richard-cox — APPROVED** [review]

**Pull request merged**